esphome/components/qmc5883l/sensor.py

```diff
@@ -103,7 +103,11 @@ def validate_enum_bound(value):
 
 
 def auto_data_rate(config):
-    interval_sec = config[CONF_UPDATE_INTERVAL].seconds
+    interval_sec = 0
+    if config[CONF_UPDATE_INTERVAL].seconds is not None:
+        interval_sec = config[CONF_UPDATE_INTERVAL].seconds
+    if config[CONF_UPDATE_INTERVAL].milliseconds is not None:
+        interval_sec += config[CONF_UPDATE_INTERVAL].milliseconds / 1000
```

Comment:
Its never None... 
Its set by line 100 above with a default of 60s

This would be better:

```suggestion
      interval_sec = config[CONF_UPDATE_INTERVAL].total_milliseconds / 1000
```

Reply:
Sure. It is great. Thank you.